### ISSUE TITLE
Multi-GPU training

### ISSUE THREAD

**Issue by longmalongma** [hkchengrex/XMem#22]
@hkchengrex Hello, thank you for your wonderful work, I don't have a single gpu with large memory, I only have 4 gpus of 12g, will I use four gpus for distributed training, will it affect the accuracy? My training code is as follows:

CUDA_VISIBLE_DEVICES=0,1,2,3 OMP_NUM_THREADS=64 python -m torch.distributed.run --master_port 1713 --nproc_per_node=4 train.py --exp_id s2 --stage 2

Is it ok?

**Comment by hkchengrex** [comment]
Should work.

**Comment by hkchengrex** [comment]
That would only train a stage 2 model (DAVIS & YouTubeVOS only) though.

**Comment by longmalongma** [comment]
> CUDA_VISIBLE_DEVICES=0,1,2,3 OMP_NUM_THREADS=64 python -m torch.distributed.run --master_port 1713 --nproc_per_node=4 train.py --exp_id s2 --stage 2

@hkchengrex 
A script trained on static images（stage 0） like this?
CUDA_VISIBLE_DEVICES=0,1,2,3 OMP_NUM_THREADS=64 python -m torch.distributed.run --master_port 1713 --nproc_per_node=4 train.py --exp_id s0 --stage 0

A script trained on static images（stage 1） like this?
CUDA_VISIBLE_DEVICES=0,1,2,3 OMP_NUM_THREADS=64 python -m torch.distributed.run --master_port 1713 --nproc_per_node=4 train.py --exp_id s1 --stage 1

By the way, the shorter training is: s0, s3, and the longer training is: s0, s1, s2, right?

**Comment by hkchengrex** [comment]
We implemented auto-staging in this repo. So you can do `--stage 03` or `--stage 012` and it will train automatically. Please see the training readme file.

**Comment by longmalongma** [comment]
> We implemented auto-staging in this repo. So you can do `--stage 03` or `--stage 012` and it will train automatically. Please see the training readme file.

Sorry, I am very confused, what is the difference between DAVIS+YoutubeVOS, longer and DAVIS+YoutubeVOS, shorter？
@hkchengrex

**Comment by hkchengrex** [comment]
One trains longer, the other trains shorter. The same was used in STCN.

**Comment by longmalongma** [comment]
> We implemented auto-staging in this repo. So you can do `--stage 03` or `--stage 012` and it will train automatically. Please see the training readme file.

@hkchengrex How should I set the stage if I want to train separately? For longer train, first, --stage 0, second, --stage 1, third, --stage 2. For shorter train, first, --stage 0, second, --stage 3. Is this correct?

**Comment by hkchengrex** [comment]
I don't know why you would do that but yes.

**Comment by longmalongma** [comment]
> I don't know why you would do that but yes.

Ok, I see, thank you very much for your reply.
